packages/block-editor/src/components/block-list/block.js

```diff
@@ -558,13 +477,266 @@ const applyWithDispatch = withDispatch( ( dispatch, ownProps, registry ) => {
+						: undefined,
+				isHighlighted: isBlockHighlighted( clientId ),
+				isMultiSelected,
+				isPartiallySelected:
+					isMultiSelected &&
+					! __unstableIsFullySelected() &&
+					! __unstableSelectionHasUnmergeableBlock(),
+				isReusable: isReusableBlock( blockType ),
+				isDragging: isBlockBeingDragged( clientId ),
+				hasChildSelected: isAncestorOfSelectedBlock,
+				removeOutline: _isSelected && outlineMode && typing,
+				isBlockMovingMode: !! movingClientId,
+				canInsertMovingBlock:
+					movingClientId &&
+					canInsertBlockType(
+						getBlockName( movingClientId ),
+						getBlockRootClientId( clientId )
+					),
+				isEditingDisabled:
+					getBlockEditingMode( clientId ) === 'disabled',
+				className: hasLightBlockWrapper
+					? attributes.className
+					: undefined,
+				defaultClassName: hasLightBlockWrapper
+					? getBlockDefaultClassName( blockName )
+					: undefined,
+			};
+		},
+		[ clientId, rootClientId ]
+	);
+
+	const {
+		mode,
+		isSelectionEnabled,
+		isLocked,
+		canRemove,
+		canMove,
+		block,
+		name,
+		attributes,
+		isValid,
+		isSelected,
+		themeSupportsLayout,
+		isTemporarilyEditingAsBlocks,
+		blockEditingMode,
+		mayDisplayControls,
+		mayDisplayParentControls,
+		index,
+		blockApiVersion,
+		blockTitle,
+		isPartOfSelection,
+		adjustScrolling,
+		enableAnimation,
+		isSubtreeDisabled,
+		isOutlineEnabled,
+		hasOverlay,
+		initialPosition,
+		isHighlighted,
+		isMultiSelected,
+		isPartiallySelected,
+		isReusable,
+		isDragging,
+		hasChildSelected,
+		removeOutline,
+		isBlockMovingMode,
+		canInsertMovingBlock,
+		isEditingDisabled,
+		className,
+		defaultClassName,
+	} = selectedProps;
+
+	// Block is sometimes not mounted at the right time, causing it be
+	// undefined see issue for more info
+	// https://github.com/WordPress/gutenberg/issues/17013
+	if ( ! selectedProps ) {
+		return null;
+	}
+
+	const publicAPI = {
+		mode,
+		isSelectionEnabled,
+		isLocked,
+		canRemove,
+		canMove,
+		// Users of the editor.BlockListBlock filter used to be able to
+		// access the block prop.
+		// Ideally these blocks would rely on the clientId prop only.
+		// This is kept for backward compatibility reasons.
+		block,
+		name,
+		attributes,
+		isValid,
+		isSelected,
+	};
+
+	const privateContext = {
+		clientId,
+		className,
+		index,
+		mode,
+		name,
+		blockApiVersion,
+		blockTitle,
+		isSelected,
+		isPartOfSelection,
+		adjustScrolling,
+		enableAnimation,
+		isSubtreeDisabled,
+		isOutlineEnabled,
+		hasOverlay,
+		initialPosition,
+		blockEditingMode,
+		isHighlighted,
+		isMultiSelected,
+		isPartiallySelected,
+		isReusable,
+		isDragging,
+		hasChildSelected,
+		removeOutline,
+		isBlockMovingMode,
+		canInsertMovingBlock,
+		isEditingDisabled,
+		isTemporarilyEditingAsBlocks,
+		defaultClassName,
+		mayDisplayControls,
+		mayDisplayParentControls,
+		themeSupportsLayout,
+	};
+
+	return (
+		<BlockListBlockContext.Provider value={ privateContext }>
```

Comment:
So this context is set here but modified within the component `BlockListBlock` and set again right? Why twice?

Reply:
Also can we set the public block edit context here (so all contexts are set in the same place)?

Reply:
Twice because we need to add information from the filters: wrapperProps and className. I'll comment on that.

Maybe we can move the public context here, but I'd prefer to try that in a separate PR.

Reply:
I added comments and explained in the description.